### PR TITLE
Refactoring ededup as its own namespace dpk_ededup

### PR DESCRIPTION
**Pull request by touma-I**
## Why are these changes needed?

First pass at restructuring the transform built as its own module (e.g. dpk_ededup) with a ray submodule (dpk_ededup.ray ).

Removed python and ray folders and refactor Dockerfile.python and Dockerfile.ray
remove pyproject.toml and Makefiles
move python code under dpk_ededup
move ray code under dpk_ededup/ray
change import statement to include module name
replace recursive Makefile and use targets from .make.cicd.targets
adapt kfp_ray/Makefile and other make target

## Related issue number (if any).

https://github.com/IBM/data-prep-kit/issues/774

### CODE REVIEW

File: transforms/universal/ededup/README.md
Comment (cmadam) [inline transforms/universal/ededup/README.md:1]: Please fix the following links:
* link to the notebook (should point to ededup_python.ipynb, currently pointing to ededup.ipynb)
* link to EdedupTransform (should point to dpk_ededup/transform_base.py, currently pointing to src/ededup_transform_python.py)

File: transforms/universal/ededup/Makefile
Comment (cmadam) [inline transforms/universal/ededup/Makefile:1]: Currently, `make run-cli-sample` does not work.
```
python -m dpk_ededup.ray.transform \
                --run_locally True --data_local_config "{ 'input_folder' : 'test-data/input', 'output_folder' : 'output'}"  \
                --ededup_num_hashes 2
Traceback (most recent call last):
  File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/cma/de/data-prep-kit-pkg/transforms/universal/ededup/dpk_ededup/ray/transform.py", line 18, in <module>
    import ray
ModuleNotFoundError: No module named 'ray'
make: *** [Makefile:18: run-cli-sample] Error 1
```
Can you please activate the virtual environment prior to launching the app?

Comment (touma-I) [reply transforms/universal/ededup/Makefile:1]: good catch. Thanks @cmadam . Done.

**Comment by touma-I** [comment]
Moved to main repo for debugging kip

**Pull request closed without merging**